### ISSUE TITLE
Removing files_sharing from whitelist breaks JS

### ISSUE THREAD

**Issue by micbar** [owncloud/guests#275]
### Steps to reproduce
1. Remove files_sharing from the whitelist
2. Create Guest User by sharing a file with him
3. Log in as Guest User
4. Try to re Share the file

### Expected behaviour
Sharing dialogue should appear

### Actual behaviour
- Endless spinning wheel shown, JS breaks
![screenshot_2018-11-27 files - owncloud enterprise edition](https://user-images.githubusercontent.com/4378513/49096323-a4321180-f26a-11e8-8011-1735d7cab2cd.png)
- Shared with you Listing is empty
![screenshot_2018-11-27 shared with you - owncloud enterprise edition](https://user-images.githubusercontent.com/4378513/49096358-b6ac4b00-f26a-11e8-9353-d4cfb462c19c.png)

### Server configuration
**Operating system**:
Ubuntu
**Web server:**
Apache
**Database:**
MySQL
**PHP version:**
7.1
**ownCloud version:** (see ownCloud admin page)
10.0.9

#### Browser log
```
TypeError: shares[0] is undefined[Weitere Informationen] sharedfilelist.js:279:4
	reloadCallback http://localhost:9680/apps/files_sharing/js/sharedfilelist.js:279:4
	reloadCallback self-hosted:978:17
	then/</</< http://localhost:9680/core/vendor/jquery/dist/jquery.min.js:2:28292
	j http://localhost:9680/core/vendor/jquery/dist/jquery.min.js:2:26920
	fireWith http://localhost:9680/core/vendor/jquery/dist/jquery.min.js:2:27738
	Deferred/</e[f[0]] http://localhost:9680/core/vendor/jquery/dist/jquery.min.js:2:28718
	j http://localhost:9680/core/vendor/jquery/dist/jquery.min.js:2:26920
	fireWith http://localhost:9680/core/vendor/jquery/dist/jquery.min.js:2:27738
	x http://localhost:9680/core/vendor/jquery/dist/jquery.min.js:4:11276
	b/< http://localhost:9680/core/vendor/jquery/dist/jquery.min.js:4:14765
```

**Comment by micbar** [comment]
@pmaier1

**Comment by PVince81** [comment]
Yeah... don't do that.

What's the use case ?

We could maybe hard-code the files_sharing into the whitelist internally to avoid errors.

**Comment by pmaier1** [comment]
> We could maybe hard-code the files_sharing into the whitelist internally to avoid errors.

Yes, we should hardcode all of the apps that should not be removed to prevent such issues. We already did that for some core components, afair!?

**Comment by micbar** [comment]
@PVince81 
This showed up because a customer wanted to prevent Guests from re-sharing.
He did it the wrong way.

The best solution is to exclude the Guest Group from sharing.
